src/Components/itemDetail.js

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import '../App.css';
+import { useParams } from 'react-router-dom';
+import { getLocalToken } from '../lib/token.js';
+import { FirestoreDocument } from 'react-firestore';
+import { Link } from 'react-router-dom';
+
+function ItemDetailPage() {
+  let { docId } = useParams();
+  // date formater
+
+  // calculate days until next purchase
+  const getFormattedDate = date => {
+    const seconds = date.seconds;
+    const year = Math.floor(seconds / 31536000);
+    const nummonths = Math.floor((seconds % 31536000) / 2628000);
+    const numdays = Math.floor(((seconds % 31536000) % 2628000) / 86400);
+
+    const month = nummonths.toString().padStart(2, '0');
+    const day = numdays.toString().padStart(2, '0');
+
+    return month + '/' + day + '/' + year;
+  };
+
+  // this is duplicated from getList lines 87-93. Should refactor in seperate issue.
+  const today = new Date();
+  const getDaysToNextPurchase = item => {
+    return Math.floor(
+      (item.nextPurchaseDate.toMillis() - today.getTime()) / (1000 * 3600 * 24),
+    );
+  };
+
+  // top navigation to go back
+  return (
+    <FirestoreDocument
+      path={`${getLocalToken()}/${docId}`}
+      render={({ isLoading, data }) => {
+        return isLoading ? (
+          <div>loading...</div>
+        ) : (
+          <div className="shadow bg-white pa2">
+            <nav className="pa1 flex">
+              <Link to="/list-view">
+                <img src="/img/arrow_back-24px.svg" alt="back"></img>
```

Comment:
Fun fact about `<img>` elements - [they don't need to have accompanying closing tags](https://stackoverflow.com/questions/23890716/why-is-the-img-tag-not-closed-in-html/23890817#:~:text=The%20tag%20represents%20what,tag%20needs%20to%20be%20closed.)!